bio2zarr/vcf.py

```diff
@@ -948,19 +948,20 @@ def init(
         return self.num_partitions
 
     def mkdirs(self):
-        # TODO add worker_processes here and do this with the ParallelWorkManager
         logger.info(
             f"Creating {len(self.metadata.fields) * self.num_partitions} directories"
         )
         self.path.mkdir()
         self.wip_path.mkdir()
-        for field in self.metadata.fields:
-            col_path = get_vcf_field_path(self.path, field)
-            logger.debug(f"Make directories for {field.full_name} at {col_path}")
-            col_path.mkdir(parents=True)
-            for j in range(self.num_partitions):
-                part_path = col_path / f"p{j}"
-                part_path.mkdir()
+        # Due to high latency batch system filesystems, we create all the directories in parallel
+        progress_config = core.ProgressConfig(total=len(self.metadata.fields) * self.num_partitions, units="dir", title="Creating directories", show=True)
+        with core.ParallelWorkManager(worker_processes=50, progress_config=progress_config) as manager:
+            for field in self.metadata.fields:
+                col_path = get_vcf_field_path(self.path, field)
+                manager.submit(col_path.mkdir, parents=True, exist_ok=True)
+                for j in range(self.num_partitions):
+                    part_path = col_path / f"p{j}"
+                    manager.submit(part_path.mkdir, exist_ok=True)
```

Comment:
I think you need ``parents=True`` here in case one of these gets done before the field directly.

I don't think we need ``exist_ok=True`` - I don't see how one of these partition dirs could already exist (unless there's an error).